### PR TITLE
refactor: add p2p headers exchange duration metrics

### PR DESCRIPTION
**Pull request by mrekucci**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2513)

### CODE REVIEW

**Review by acud — APPROVED** [review]

File: pkg/p2p/libp2p/libp2p.go [hunk 1]
```diff
@@ -482,6 +483,7 @@ func (s *Service) AddProtocol(p p2p.ProtocolSpec) (err error) {
 				_ = stream.Reset()
 				return
 			}
+			s.metrics.HeadersExchangeDuration.Observe(float64(time.Since(start).Nanoseconds()))
```
Comment (istae) [inline pkg/p2p/libp2p/libp2p.go:486]: time.Since(start).Seconds() returns float64 and make more sense here instead of Nanoseconds

Comment (istae) [reply pkg/p2p/libp2p/libp2p.go:486]: The default histograms buckets also expect seconds I believe

Comment (istae) [reply pkg/p2p/libp2p/libp2p.go:486]:
```
// DefBuckets are the default Histogram buckets. The default buckets are
// tailored to broadly measure the response time (in seconds) of a network
// service. Most likely, however, you will be required to define buckets
// customized to your use case.
var (
	DefBuckets = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10}

	errBucketLabelNotAllowed = fmt.Errorf(
		"%q is not allowed as label name in histograms", bucketLabel,
	)
)
```

Comment (istae) [reply pkg/p2p/libp2p/libp2p.go:486]:
```
// DefBuckets are the default Histogram buckets. The default buckets are
// tailored to broadly measure the response time (in seconds) of a network
// service. Most likely, however, you will be required to define buckets
// customized to your use case.
var (
	DefBuckets = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10}

	errBucketLabelNotAllowed = fmt.Errorf(
		"%q is not allowed as label name in histograms", bucketLabel,
	)
)
```

**Review by istae — CHANGES_REQUESTED** [review]

File: pkg/p2p/libp2p/libp2p.go (hunk 1 above)
Comment (istae) [reply pkg/p2p/libp2p/libp2p.go:486]:
```
// DefBuckets are the default Histogram buckets. The default buckets are
// tailored to broadly measure the response time (in seconds) of a network
// service. Most likely, however, you will be required to define buckets
// customized to your use case.
var (
	DefBuckets = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10}

	errBucketLabelNotAllowed = fmt.Errorf(
		"%q is not allowed as label name in histograms", bucketLabel,
	)
)
```

**Review by mrekucci — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2513)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @istae and @janos)

___
*[pkg/p2p/libp2p/libp2p.go, line 486 at r1](https://reviewable.io/reviews/ethersphere/bee/2513#-MjhxYWrghMnD7eCqyA_-r1-486:-MjiF5WM3pesitrX-jLp:bhmhk5s) ([raw file](https://github.com/ethersphere/bee/blob/83086d2994818feb1dffbbf8bf38806a235a063a/pkg/p2p/libp2p/libp2p.go#L486)):*
<details><summary><i>Previously, istae (istae) wrote…</i></summary><blockquote>

```
// DefBuckets are the default Histogram buckets. The default buckets are
// tailored to broadly measure the response time (in seconds) of a network
// service. Most likely, however, you will be required to define buckets
// customized to your use case.
var (
	DefBuckets = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10}

	errBucketLabelNotAllowed = fmt.Errorf(
		"%q is not allowed as label name in histograms", bucketLabel,
	)
)
```
</blockquote></details>

Nice catch! Done.

**Review by istae — APPROVED** [review]

**Review by acud — APPROVED** [review]

**Pull request merged**